src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -345,6 +345,10 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
       case (None, Some(b))    => Some(DuplexConsensusRead(id=b.id, b.bases, b.quals, b.errors, b, None))
       case (Some(a), Some(b)) =>
         val len = min(a.length, b.length)
+        if (a.depths.take(len).forall(_ == 0)) {
+          require(b.depths.take(len).exists(_ > 0), "Bug: both consensus reads had zero depths across all bases")
+          return duplexConsensus(ba, ab, sourceReads)
+        }
```

Comment:
I don't like returns, but I don't see a way to structure the match better without some ugly refactoring.  Thoughts?

Reply:
Alternatively, we could change all single strand consensus reads that are all Ns to `None` we call `duplexConsensus`?